### Task 1 - Import the data into 2d array format and display the number of rows and columns 

In [7]:
admission_data = np.genfromtxt('DS1_C4_S1_AdmissionPredict_Data_Practice.csv',delimiter=',',skip_header = 1)

In [8]:
import numpy as np

In [9]:
admission_data

array([[  1.  , 337.  , 118.  , ...,   9.65,   1.  ,   0.92],
       [  2.  , 324.  , 107.  , ...,   8.87,   1.  ,   0.76],
       [  3.  , 316.  , 104.  , ...,   8.  ,   1.  ,   0.72],
       ...,
       [398.  , 330.  , 116.  , ...,   9.45,   1.  ,   0.91],
       [399.  , 312.  , 103.  , ...,   8.78,   0.  ,   0.67],
       [400.  , 333.  , 117.  , ...,   9.66,   1.  ,   0.95]])

In [10]:
admission_data.shape

(400, 9)

In [11]:
admission_data.ndim

2

Interpretation - Here i have imported the file in a 2d array format and i have displayed the number of rows and columns present in the data .

### Task 2 - Display the GRE score of all students 

In [12]:
GRE_scores = admission_data[:,1]
print(GRE_scores)


[337. 324. 316. 322. 314. 330. 321. 308. 302. 323. 325. 327. 328. 307.
 311. 314. 317. 319. 318. 303. 312. 325. 328. 334. 336. 340. 322. 298.
 295. 310. 300. 327. 338. 340. 331. 320. 299. 300. 304. 307. 308. 316.
 313. 332. 326. 322. 329. 339. 321. 327. 313. 312. 334. 324. 322. 320.
 316. 298. 300. 311. 309. 307. 304. 315. 325. 325. 327. 316. 318. 328.
 332. 336. 321. 314. 314. 329. 327. 301. 296. 294. 312. 340. 320. 322.
 340. 319. 315. 317. 314. 316. 318. 299. 298. 301. 303. 304. 306. 331.
 332. 323. 322. 312. 314. 317. 326. 316. 329. 338. 331. 304. 305. 321.
 301. 320. 311. 310. 299. 290. 296. 327. 335. 334. 310. 308. 301. 300.
 323. 319. 326. 333. 339. 303. 309. 323. 333. 314. 312. 316. 326. 318.
 329. 332. 331. 340. 325. 320. 315. 326. 339. 311. 334. 332. 321. 324.
 326. 312. 315. 309. 306. 297. 315. 298. 318. 317. 329. 322. 302. 313.
 293. 311. 312. 334. 322. 323. 321. 320. 329. 319. 309. 307. 300. 305.
 299. 314. 316. 327. 317. 335. 331. 324. 324. 323. 322. 336. 316. 307.
 306. 

Interpretation - Here i have displayed the GRE_scores of all students

### Task 3 - The total number of students who have GRE_scores greater than 320 

In [13]:
GRE = GRE_scores>320

In [14]:
filtered_op = GRE_scores[GRE]
filtered_op

array([337., 324., 322., 330., 321., 323., 325., 327., 328., 325., 328.,
       334., 336., 340., 322., 327., 338., 340., 331., 332., 326., 322.,
       329., 339., 321., 327., 334., 324., 322., 325., 325., 327., 328.,
       332., 336., 321., 329., 327., 340., 322., 340., 331., 332., 323.,
       322., 326., 329., 338., 331., 321., 327., 335., 334., 323., 326.,
       333., 339., 323., 333., 326., 329., 332., 331., 340., 325., 326.,
       339., 334., 332., 321., 324., 326., 329., 322., 334., 322., 323.,
       321., 329., 327., 335., 331., 324., 324., 323., 322., 336., 340.,
       334., 325., 328., 338., 333., 331., 330., 322., 321., 324., 324.,
       324., 330., 326., 325., 329., 324., 325., 328., 324., 321., 335.,
       321., 324., 326., 331., 327., 324., 325., 327., 322., 329., 321.,
       340., 331., 336., 324., 330., 322., 323., 321., 323., 325., 328.,
       324., 327., 323., 326., 324., 327., 325., 332., 323., 324., 326.,
       325., 327., 321., 322., 334., 338., 330., 32

In [15]:
np.count_nonzero(filtered_op)

163

Interpretation - Here is the output for the students whose GRE_scores are greater than 320 and the total number of students whose GRE_scores are greater than 320 is 163

### Task 4 - Chance of admission of student whose TOEFL score is greater than 110 

In [16]:
TOEFL_score = admission_data[:,2]
TOEFL_score

array([118., 107., 104., 110., 103., 115., 109., 101., 102., 108., 106.,
       111., 112., 109., 104., 105., 107., 106., 110., 102., 107., 114.,
       116., 119., 119., 120., 109.,  98.,  93.,  99.,  97., 103., 118.,
       114., 112., 110., 106., 105., 105., 108., 110., 105., 107., 117.,
       113., 110., 114., 119., 110., 111.,  98., 100., 116., 112., 110.,
       103., 102.,  99.,  99., 104., 100., 101., 105., 107., 111., 112.,
       114., 107., 109., 115., 118., 112., 111., 108., 106., 114., 112.,
        99.,  95.,  93., 105., 120., 110., 115., 115., 103., 106., 107.,
       108., 109., 106.,  97.,  98.,  97.,  99., 100., 100., 120., 119.,
       113., 107., 105., 106., 104., 112., 110., 111., 117., 116., 103.,
       108., 109., 107., 110., 105., 106., 102., 104.,  99., 104., 117.,
       119., 106., 108., 106., 100., 113., 112., 112., 118., 114., 105.,
       105., 112., 113., 109., 103., 100., 116., 109., 110., 118., 115.,
       120., 112., 113., 105., 114., 116., 106., 11

In [17]:
TOEFL = TOEFL_score>110
filtered_op = TOEFL_score[TOEFL]
filtered_op

array([118., 115., 111., 112., 114., 116., 119., 119., 120., 118., 114.,
       112., 117., 113., 114., 119., 111., 116., 112., 111., 112., 114.,
       115., 118., 112., 111., 114., 112., 120., 115., 115., 120., 119.,
       113., 112., 111., 117., 116., 117., 119., 113., 112., 112., 118.,
       114., 112., 113., 116., 118., 115., 120., 112., 113., 114., 116.,
       114., 116., 112., 111., 117., 113., 111., 111., 119., 113., 118.,
       115., 112., 111., 114., 118., 120., 120., 120., 119., 117., 116.,
       112., 113., 112., 111., 113., 111., 112., 114., 115., 114., 111.,
       115., 114., 119., 111., 113., 113., 111., 112., 116., 118., 114.,
       120., 114., 112., 112., 111., 113., 116., 112., 113., 111., 118.,
       116., 115., 114., 119., 113., 117., 112., 111., 116., 117.])

In [92]:
admission_chance = admission_data[TOEFL]
admission_chance[:,8]*100


array([92., 90., 84., 78., 70., 94., 95., 97., 94., 91., 90., 94., 87.,
       91., 86., 89., 78., 78., 72., 52., 55., 61., 78., 94., 96., 93.,
       72., 74., 96., 92., 94., 86., 90., 79., 74., 87., 91., 93., 94.,
       94., 85., 78., 84., 92., 96., 79., 89., 80., 90., 92., 97., 80.,
       81., 83., 96., 93., 94., 86., 81., 89., 89., 87., 85., 90., 89.,
       93., 93., 88., 90., 86., 94., 97., 97., 95., 96., 94., 93., 91.,
       76., 71., 82., 91., 88., 85., 86., 70., 76., 77., 93., 85., 90.,
       70., 83., 89., 80., 94., 93., 92., 89., 86., 90., 71., 80., 79.,
       80., 81., 80., 80., 83., 94., 93., 91., 86., 95., 96., 96., 84.,
       89., 91., 95.])

In [96]:
index = admission_chance[:,0]

In [97]:
index

array([  1.,   6.,  12.,  13.,  22.,  23.,  24.,  25.,  26.,  33.,  34.,
        35.,  44.,  45.,  47.,  48.,  50.,  53.,  54.,  65.,  66.,  67.,
        70.,  71.,  72.,  73.,  76.,  77.,  82.,  84.,  85.,  98.,  99.,
       100., 105., 107., 108., 109., 121., 122., 127., 128., 129., 130.,
       131., 134., 135., 139., 142., 143., 144., 145., 146., 148., 149.,
       151., 152., 153., 165., 172., 174., 175., 176., 177., 186., 188.,
       189., 190., 191., 193., 194., 203., 204., 213., 214., 215., 216.,
       217., 223., 229., 230., 235., 236., 237., 238., 243., 244., 250.,
       254., 255., 260., 264., 269., 277., 284., 285., 286., 287., 288.,
       298., 299., 300., 308., 319., 320., 326., 329., 331., 336., 338.,
       362., 363., 366., 373., 385., 386., 393., 395., 398., 400.])

Interpretation - Here i have displayed the chance of admission in percentage where TOEFL_score is greater than 110 and the index number of those students

### Task 5 - Find the students who are eligible for the scholarship 

In [41]:
Research = admission_data[:,7]

In [42]:
Research

array([1., 1., 1., 1., 0., 1., 1., 0., 0., 0., 1., 1., 1., 1., 1., 0., 0.,
       1., 0., 0., 1., 0., 1., 1., 1., 1., 0., 1., 0., 0., 1., 1., 1., 1.,
       1., 1., 0., 0., 0., 0., 1., 1., 1., 0., 1., 1., 1., 0., 1., 1., 1.,
       1., 1., 1., 0., 0., 0., 0., 1., 0., 0., 0., 1., 1., 0., 0., 0., 1.,
       1., 1., 1., 1., 1., 1., 0., 1., 1., 0., 1., 0., 1., 1., 1., 1., 1.,
       0., 0., 0., 0., 1., 1., 0., 0., 1., 0., 0., 0., 1., 1., 1., 1., 0.,
       0., 0., 1., 1., 1., 1., 1., 0., 0., 1., 1., 0., 1., 1., 0., 0., 0.,
       1., 1., 1., 0., 0., 0., 1., 1., 1., 1., 1., 1., 0., 0., 0., 1., 1.,
       0., 1., 1., 0., 1., 1., 1., 1., 1., 1., 0., 1., 1., 1., 1., 1., 1.,
       0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 1., 0., 0., 0., 1., 0.,
       1., 1., 1., 1., 1., 1., 1., 0., 0., 0., 0., 0., 0., 0., 0., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 0., 1., 0., 0., 0., 0., 1., 1., 1., 1.,
       0., 0., 0., 1., 0., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 0., 0.,
       0., 0., 0., 0., 0.

In [51]:
scholar = np.logical_and.reduce((GRE_scores > 315, TOEFL_score > 115, Research))
filtered_op = admission_data[scholar]

In [52]:
filtered_op

array([[  1.  , 337.  , 118.  ,   4.  ,   4.5 ,   4.5 ,   9.65,   1.  ,
          0.92],
       [ 23.  , 328.  , 116.  ,   5.  ,   5.  ,   5.  ,   9.5 ,   1.  ,
          0.94],
       [ 24.  , 334.  , 119.  ,   5.  ,   5.  ,   4.5 ,   9.7 ,   1.  ,
          0.95],
       [ 25.  , 336.  , 119.  ,   5.  ,   4.  ,   3.5 ,   9.8 ,   1.  ,
          0.97],
       [ 26.  , 340.  , 120.  ,   5.  ,   4.5 ,   4.5 ,   9.6 ,   1.  ,
          0.94],
       [ 33.  , 338.  , 118.  ,   4.  ,   3.  ,   4.5 ,   9.4 ,   1.  ,
          0.91],
       [ 53.  , 334.  , 116.  ,   4.  ,   4.  ,   3.  ,   8.  ,   1.  ,
          0.78],
       [ 71.  , 332.  , 118.  ,   5.  ,   5.  ,   5.  ,   9.64,   1.  ,
          0.94],
       [ 82.  , 340.  , 120.  ,   4.  ,   5.  ,   5.  ,   9.5 ,   1.  ,
          0.96],
       [ 98.  , 331.  , 120.  ,   3.  ,   4.  ,   4.  ,   8.96,   1.  ,
          0.86],
       [ 99.  , 332.  , 119.  ,   4.  ,   5.  ,   4.5 ,   9.24,   1.  ,
          0.9 ],
       [108.  , 338. 

In [63]:
total_no_of_eligible_for_scholar = len(filtered_op)
total_no_of_eligible_for_scholar

41

In [99]:
index = filtered_op[:,0]
index

array([  1.,  23.,  24.,  25.,  26.,  33.,  53.,  71.,  82.,  98.,  99.,
       108., 109., 121., 122., 130., 139., 142., 144., 149., 152., 172.,
       177., 188., 194., 203., 204., 213., 214., 215., 216., 260., 286.,
       287., 326., 338., 362., 373., 386., 398., 400.])

Interpretation - These are the index of the students who are eligible for the scholarship and the total number of student eligible for scholar is 41